src/clients/apiClient.ts

```diff
@@ -123,297 +126,74 @@ export async function createNewUser(newUser: NewUser): Promise<Response> {
   }
 }
 
-// export const getWhaleSightings = async (): Promise<NewSighting[]> => {
-//   let response: Response;
-//   response = await fetch(`https://${backendUrl}/sightings/`);
-//   return response;
-// }
+export interface SpeciesSearch {
+  tailType: number;
+  size: number;
+  colour: string;
+}
+
+export async function fetchSpeciesQuery(speciesSearch: SpeciesSearch): Promise<Response> {
+  const response = await fetch(`${backendUrl}/species?TailType=${speciesSearch.tailType}&Size=${speciesSearch.size}&Colour=${speciesSearch.colour}`);
+  if (!response.ok) {
+    throw new Error(await response.json());
+  } else {
+    return await response.json();
+  }
+}
 
-export async function getWhaleSightings(): Promise<NewSighting[]> {
-  return new Promise<NewSighting[]>((resolve, reject) => {
-    const sightings = newSightings;
-    if (sightings) {
-      resolve(sightings);
+export async function fetchSightingById(sightingId: number): Promise<WhaleSighting> {
+  const response = await fetch(`${backendUrl}/sightings/${sightingId}`);
+  if (!response.ok) {
+    throw new Error(await response.json());
+  }
+  else {
+    return await response.json();
+  }
+}
+
+export async function deleteLike(likeId: number): Promise<Response> {
+  const response = await fetch(`${backendUrl}/likes/delete/${likeId}`);
+  if (!response.ok) {
+    throw new Error(await response.json());
+  }
+  else {
+    return response;
+  }
+}
+
+export async function fetchAllApprovedSightings(): Promise<WhaleSighting[]> {
+  const response = await fetch(`${backendUrl}/sightings`);
+  
+  if (!response.ok) {
+    throw new Error(await response.json());
+  }
+  else {
+    return await response.json();
+  }
+}
+
+export async function createLike(newLike: NewLike): Promise<Response> {
+  const response = await fetch(`${backendUrl}/likes/create`, {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json"
+    },
+    body: JSON.stringify(newLike),
+  });
+  if (!response.ok) {
+    throw new Error(await response.json());
+  }
+  else {
+    return await response.json();
+  }
+}
+
+export async function isLikedByUser(sightingId: number): Promise<boolean> {
+  return new Promise<boolean>((resolve, reject) => {
+    if (true) {
+      resolve(true);
     } else {
-      reject("Failed to get sighting data");
+      reject(false);
```

Comment:
Are we using this method or will this just be returned in the `fetchAllApprovedSightings` response?